packages/esm-service-queues-app/src/helpers/useQueues.ts

```diff
@@ -0,0 +1,13 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type QueueServiceInfo } from '../types';
+
+export function useQueues(locationUuid?: string) {
+  const apiUrl = '/ws/rest/v1/queue' + (locationUuid ? `?location=${locationUuid}` : '');
+
+  const { data } = useSWRImmutable<{ data: { results: Array<QueueServiceInfo> } }, Error>(apiUrl, openmrsFetch);
```

Comment:
Do we expect queues to be [immutable](https://swr.vercel.app/docs/revalidation.en-US#disable-automatic-revalidations) resources? If not, you could consider enabling automatic revalidations for the resource by using the useSWR hook instead.

Reply:
I believe so. Queues almost never change.